editorextensions/codebeamer-custom-ui/modal/src/api/lucidGateway.ts

```diff
@@ -5,11 +5,77 @@ import { store } from '../store/store';
 import TrackerDetails from '../models/trackerDetails.if';
 import { CardData } from '../models/lucidCardData';
 
+/**
+ * Class for handling message events and callbacks.
+ */
+export class MessageHandler {
+	private callbacks: ((data: any) => void)[] = [];
+
+	/**
+	 * Creates an instance of MessageHandler and sets up a message event listener.
+	 */
+	constructor() {
+		window.addEventListener('message', (e) => {
+			const data = JSON.parse(e.data);
+			this.notifyCallbacks(data);
+		});
+	}
+
+	/**
+	 * Requests card blocks from the parent window and registers a callback to handle the response.
+	 * @param {function} callback - The callback function that will be called with the received card block data.
+	 */
+	getCardBlocks(callback: (arg0: any) => void) {
+		this.subscribeCallback(callback);
+
+		window.parent.postMessage(
+			{ action: 'getCardBlocks', payload: {} },
+			'*'
+		);
+	}
```

Comment:
We're slightly starting to get responsibilities mixed up here, and would do good to make things clear & separated off the start:
This abstract method per se is neat for any consumer of the `MessageHandler`, but it shouldn't by itself make the `postMessage` call. We only want to _handle_ messages in here, not send them too per se.  
Let's contain _all_ `postMessage` logic in the (de facto) `LucidGateway` _class_ (not _file_), which is currently just the loose arrangement of individual functions _in_ the `lucidGateway.ts` _file_ (excluding the `MessageHandler` class).

There's different ways you could go about this. Feel free to take this a step further too, if something comes to your mind.